### DISCUSSION TITLE
내 리뷰 목록 조회 API 명세

### DISCUSSION THREAD

**Discussion by ohseil** [team-reviewing/reviewing-backend#12] [API 명세]
# 내 리뷰 목록 조회   

## 기본 정보   
```http
GET /reviews HTTP/1.1
Host: ${ HOSTNAME }
Authorization: Bearer ${ ACCESS_TOKEN }
```

### Header
|Name|Description|Required|
|-|-|-|
|Authorization|access token|O|

### Query Parameter
|Name|Type|Description|Required|
|-|-|-|-|
|role|String|리뷰어의 입장에서 요청받은 리뷰 목록을 원하는 경우 reviewer, 리뷰이의 입장에서 요청한 리뷰 목록을 원하는 경우 reviewee|O|
|status|String|리뷰 상태. 이 파라미터가 포함되지 않으면 모든 상태를 조회한다.|X|

## Response

|Name|Type|Description|Required|
|-|-|-|-|
|reviews|ReviewSummary[]|리뷰 요약 정보 목록|X|

### ReviewSummary
|Name|Type|Description|Required|
|-|-|-|-|
|id|Long|리뷰 id|O|
|title|String|리뷰 제목|O|
|reviewerId|Long|리뷰어 id|O|
|status|String|리뷰 상태|O|
|member|MemberSummary|리뷰어의 입장에서 목록을 요청하는 경우에는 리뷰이 요약 정보, 리뷰이의 입장에서 목록을 요청하는 경우에는 리뷰어 요약 정보|O|

### MemberSummary
|Name|Type|Description|Required|
|-|-|-|-|
|id|Long|유저 id|O|
|username|String|유저 이름|O|
|imageUrl|String|유저 프로필 이미지|O|

### Request
```http
GET hostname/reviews?role=reviewer
Authorization: Bearer access_token
```
```http
GET hostname/reviews?role=reviewee
Authorization: Bearer access_token
```
```http
GET hostname/reviews?role=reviewer&status=CREATED
Authorization: Bearer access_token
```

### Response
```http
HTTP/1.1 OK
Content-Type: application/json;charset=UTF-8
{
   "reviews" : [
      {
         "id" : 1,
         "title" : "제목1",
         "reviewerId" : 1,
         "status" : "CREATED",
         "member" : {
            "id" : 1,
            "username" : "유저이름1",
            "imageUrl" : "imageUrl1",
         },
      },
      {
         "id" : 2,
         "title" : "제목2",
         "reviewerId" : 2,
         "status" : "ACCEPTED",
         "member" : {
            "id" : 2,
            "username" : "유저이름2",
            "imageUrl" : "imageUrl2",
         },
      },
      {
         "id" : 3,
         "title" : "제목3",
         "reviewerId" : 3,
         "status" : "APPROVED",
         "member" : {
            "id" : 3,
            "username" : "유저이름3",
            "imageUrl" : "imageUrl3",
         },
      },
      ...
   ]
}
```

**Comment by leeuihyun** [comment]
확인했습니다~

**Comment by ohseil** [comment]
리뷰 상태 정보 추가했습니다!
리뷰 상태 정 보 명세 : #17

**Reply by leeuihyun** [reply]
상태는 일단 총 3개인가요 ?

**Reply by ohseil** [reply]
네 개인적으로 생각했을 때는 지금은 이 3개의 상태만 필요하다고 생각했습니다.
이후에 회의해서 리뷰 진행 과정이 더 구체화가 되어서 기획이 추가가 되면 상태가 추가되거나 변경될 수 있다고 생각합니다.
혹시 다른 의견 있으시면 말씀해주세요!

**Reply by leeuihyun** [reply]
확인했습니다~
